src/main/scala/mesosphere/marathon/MarathonModule.scala

```diff
@@ -143,20 +149,23 @@ class MarathonModule(conf: MarathonConf, http: HttpConf, zk: ZooKeeperClient)
       "MarathonScheduler")
   }
 
+  @Named(ModuleNames.NAMED_HOST_PORT)
+  @Provides
+  @Singleton
+  def provideHostPort: String = {
+    "%s:%d".format(conf.hostname(), http.httpPort())
+  }
+
   @Named(ModuleNames.NAMED_CANDIDATE)
   @Provides
   @Singleton
-  def provideCandidate(zk: ZooKeeperClient): Option[Candidate] = {
+  def provideCandidate(zk: ZooKeeperClient, @Named(ModuleNames.NAMED_HOST_PORT) hostPort: String): Option[Candidate] = {
     if (conf.highlyAvailable()) {
-      log.info("Registering in Zookeeper with hostname:"
-        + conf.hostname())
-      val candidate = new CandidateImpl(new ZGroup(zk, ZooDefs.Ids.OPEN_ACL_UNSAFE,
-        conf.zooKeeperLeaderPath),
+      log.info("Registering in Zookeeper with hostPort:" + hostPort)
+      val candidate = new CandidateImpl(new ZGroup(zk, ZooDefs.Ids.OPEN_ACL_UNSAFE, conf.zooKeeperLeaderPath),
```

Comment:
Side note: call parentheses either remove from the calling side or add to the caller side.